test/functional/bundleremove/remove-recursive.bats

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env bats
+
+# Author: Castulo Martinez
+# Email: castulo.martinez@intel.com
+
+load "../testlib"
+
+test_setup() {
+
+	create_test_environment "$TEST_NAME"
+	# all bundles will be marked as "installed" but only
+	# bundles test-bundle1 and test-bundle6 will be marked as "tracked"
+	create_bundle -L -t -n test-bundle1 -f /foo/test-file1 "$TEST_NAME"
+	create_bundle -L    -n test-bundle2 -f /bar/test-file2 "$TEST_NAME"
+	create_bundle -L    -n test-bundle3 -f /bat/test-file3 "$TEST_NAME"
+	create_bundle -L    -n test-bundle4 -f /baz/test-file4 "$TEST_NAME"
+	create_bundle -L    -n test-bundle5 -f /test-file5     "$TEST_NAME"
+	create_bundle -L -t -n test-bundle6 -f /test-file6     "$TEST_NAME"
+	# test-bundle1 has 4 direct dependencies: os-core, test-bundle2,
+	# test-bundle4 (also-add), test-bundle6 (tracked)
+	add_dependency_to_manifest -p    "$WEBDIR"/10/Manifest.test-bundle1 os-core
+	add_dependency_to_manifest -p    "$WEBDIR"/10/Manifest.test-bundle1 test-bundle2
+	add_dependency_to_manifest -p -o "$WEBDIR"/10/Manifest.test-bundle1 test-bundle4
+	add_dependency_to_manifest       "$WEBDIR"/10/Manifest.test-bundle1 test-bundle6
+	# and 1 indirect dependency: test-bundle3
+	add_dependency_to_manifest "$WEBDIR"/10/Manifest.test-bundle2 test-bundle3
```

Comment:
Sorry what is indirect dependency?

Reply:
bundles that are added to the system because they are included or also-added by another included or also-added bundle. 

So in the specific case of that test, test-bundle1 includes test-bundle2 (direct), and test-bundle2 includes test-bundle3 (direct for test-bundle2 but indirect for test-bundle1)

Reply:
Ahh I see..Thanks for explaining